workbench-core/authorization/src/caslAuthorizationPlugin.ts

```diff
@@ -23,12 +25,36 @@ export default class CASLAuthorizationPlugin implements AuthorizationPlugin {
       throw new ForbiddenError(err.message);
     }
   }
+
+  public async isAuthorizedOnDynamicOperations(
+    identityPermissions: IdentityPermission[],
+    dynamicOperations: DynamicOperation[]
+  ): Promise<void> {
+    const ability: Ability = this._defineAbilitiesForIdentityPermissions(identityPermissions);
+    try {
+      dynamicOperations.forEach((dynamicOperation: DynamicOperation) => {
+        const attributes = {
+          subjectId: dynamicOperation.subjectId,
+          ...dynamicOperation.subjectAttributes
+        };
+        const requestedSubject = subject(dynamicOperation.subjectType, attributes);
+        CASLForbiddenError.from(ability).throwUnlessCan(
+          dynamicOperation.action,
+          requestedSubject,
+          dynamicOperation.field
+        );
+      });
+    } catch (err) {
+      throw new ForbiddenError(err.message);
```

Comment:
I agree but I think the better option is to log it in the service

Reply:
after thinking about it more, i think this could be an anti-pattern. We are logging and throwing. This is rather inefficient because the handler could also log it again. We should just leave it to the handler to proper log it.

Reply:
Is the expectation that people will be catching this error then? An example of how its supposed to work may be helpful, because it feels a little confusing that we're throwing an Auth error that the user has the option of ignoring.

Reply:
Yes that is the expectation. I will document more of this in a dynamicAuthz README. The idea here is we want to throw an error so regardless of what the consumer wants to do with the error, they must handle it. If they choose to handle the error by ignoring the error, the consumer has made that decision knowingly. Could jump on a call so we can talk about this further.

Reply:
Sounds good. I'll find some time on your calendar tomorrow!